compiler/rustc_errors/src/lib.rs

```diff
@@ -1731,25 +1731,80 @@ impl DelayedDiagnostic {
 
 #[derive(Copy, PartialEq, Eq, Clone, Hash, Debug, Encodable, Decodable)]
 pub enum Level {
+    /// For bugs in the compiler. Manifests as an ICE (internal compiler error) panic.
+    ///
+    /// Its `EmissionGuarantee` is `BugAbort`.
     Bug,
+
+    /// This is a strange one: lets you register an error without emitting it. If compilation ends
+    /// without any other errors occurring, this will be emitted as an error. Otherwise, it will be
```

Comment:
> this will be emitted as an error

It will be emitted as a bug, I though?

Reply:
We have some unclear terminology. Sometimes "error" means `Level::Error` and sometimes it means `Level::{BugDelayed,Fatal,Error,FailureNote}`, e.g. [here](https://github.com/rust-lang/rust/blob/398fd92e177cd1a47ae1bb146506953e87e73d30/compiler/rustc_errors/src/diagnostic.rs#L263-L279).

But even with that in mind, this is a clumsy sentence, I'll fix it up in a minute.